### PR TITLE
Problem: thirdparty protobuf files are copy-pasted

### LINKED ISSUE #383: Problem: thirdparty protobuf files are copy-pasted
It would be easier to maintain if we include thirdparty(cosmos-sdk) as submodule.

### PR DESCRIPTION
**Pull request by yihuang**
fix #383

Solution:
- add submodule, change related scripts

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-com/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-com/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-com/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-com/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: x/supply/types/query.pb.gw.go [hunk 1]
```diff
@@ -72,22 +70,19 @@ func local_request_Query_LiquidSupply_0(ctx context.Context, marshaler runtime.M
...
 // StreamingRPC :currently unsupported pending https://github.com/grpc/grpc-go/issues/906.
-// Note that using this registration option will cause many gRPC library features to stop working. Consider using RegisterQueryHandlerFromEndpoint instead.
+// Note that using this registration option will cause many gRPC library features (such as grpc.SendHeader, etc) to stop working. Consider using RegisterQueryHandlerFromEndpoint instead.
 func RegisterQueryHandlerServer(ctx context.Context, mux *runtime.ServeMux, server QueryServer) error {
 
 	mux.Handle("GET", pattern_Query_TotalSupply_0, func(w http.ResponseWriter, req *http.Request, pathParams map[string]string) {
 		ctx, cancel := context.WithCancel(req.Context())
 		defer cancel()
-		var stream runtime.ServerTransportStream
-		ctx = grpc.NewContextWithServerTransportStream(ctx, &stream)
```
Comment (tomtau) [inline x/supply/types/query.pb.gw.go:82]: these aren't needed?

Comment (tomtau) [reply x/supply/types/query.pb.gw.go:82]: @devashishdxt

Comment (devashishdxt) [reply x/supply/types/query.pb.gw.go:82]: These files are auto-generated by protobuf compiler.

File: pystarport/pystarport/convert.sh [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
 TMP=$(whereis grpc_python_plugin)
 PLUGIN="$(cut -d' ' -f2 <<<"$TMP")"
 mkdir $OUTPUT
-git clone --branch v0.41.0 https://github.com/cosmos/cosmos-sdk.git
 git clone --branch v0.34.3 https://github.com/tendermint/tendermint.git
-cp -Rf $COSMOS/third_party/proto/* $COSMOS/proto/ 
-rm -rf $COSMOS/proto/cosmos/base/tendermint
 # cosmos
-protoc --proto_path=$COSMOS/proto --proto_path=$COSMOS/third_party/proto --python_out=$OUTPUT $(find $COSMOS/proto/cosmos -iname "*.proto") --grpc_python_out=$OUTPUT  --plugin=protoc-gen-grpc_python=$PLUGIN
+python -m grpc.tools.protoc --proto_path=$COSMOS/proto --proto_path=$COSMOS/third_party/proto --python_out=$OUTPUT $(find $COSMOS/proto/cosmos -iname "*.proto") --grpc_python_out=$OUTPUT  --plugin=protoc-gen-grpc_python=$PLUGIN
```
Comment (devashishdxt) [inline pystarport/pystarport/convert.sh:10]: Do we need to install some python library to run this? If yes, maybe edit readme with required dependencies.

File: x/supply/types/query.pb.gw.go (hunk 1 above)
Comment (yihuang) [reply x/supply/types/query.pb.gw.go:82]: yes, I installed the version specified in cosmos repo, maybe there's upgrades.

File: pystarport/pystarport/convert.sh (hunk 2 above)
Comment (yihuang) [reply pystarport/pystarport/convert.sh:10]: this script is normally run by `nix-shell` automatically, the environment is setup in [here](https://github.com/crypto-org-chain/chain-main/pull/384/files#diff-5819221f7ad475c47c88c3293eb44a46586748a892d982474398550247f63b13R4)

File: go.mod [hunk 3]
```diff
@@ -29,4 +29,4 @@ replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alp
 
 replace github.com/cosmos/ledger-cosmos-go => github.com/crypto-com/ledger-cosmos-go v0.9.10-0.20200929055312-01e1d341de0f
 
-replace github.com/cosmos/cosmos-sdk => github.com/crypto-com/cosmos-sdk v0.41.0-patched-tm
+replace github.com/cosmos/cosmos-sdk => ./third_party/cosmos-sdk
```
Comment (allthatjazzleo) [inline go.mod:32]: it points to master branch?

Comment (yihuang) [reply go.mod:32]: it's the same tag, the commit hash can be reviewed [here](https://github.com/crypto-org-chain/chain-main/pull/384/files#diff-000ed18c511ee5e37e33a1aeb8fa9a2ab9a9b149dcec5fc3e58f36f845ebc610)

File: x/supply/types/query.pb.gw.go (hunk 1 above)
Comment (tomtau) [reply x/supply/types/query.pb.gw.go:82]: is supply module's GRPC query tested in integration tests?
I understand this is auto-generated, but to be sure this doesn't produce invalid code / it'd crash (e.g. due to mismatch between the protobuf generator and the Go dependency library)

**Comment by tomtau** [comment]
simulation scripts may need an update too?

**Comment by allthatjazzleo** [comment]
> ❌ `/runsim` simulation test has failed 😅
> Can further check [here](https://github.com/crypto-org-chain/chain-main/actions/runs/595713519)

since event triggered pipeline will use the default branch workflow, git-submodule in PR workflow is not executed
@yihuang

File: x/supply/types/query.pb.gw.go (hunk 1 above)
Comment (yihuang) [reply x/supply/types/query.pb.gw.go:82]: I think supply module is not tested in integration test, we can do that in another issue.
Test cli query should be good, cli query calls the grpc api.

**Comment by allthatjazzleo** [comment]
maybe create a separate PR to fix workflow with submodule first
and another PR related to non workflow changes after previous merged then run `/runsim`?

File: x/supply/types/query.pb.gw.go (hunk 1 above)
Comment (yihuang) [reply x/supply/types/query.pb.gw.go:82]: I just did a manual simple test on cli, it do works.
```
$ pystarport serve --config examples/devnet.yaml
```
```
$ chain-maind q supply liquid
supply:
- amount: "40000000000"
  denom: basecro
```
```
$ chain-maind q supply total
supply:
- amount: "62000003831"
  denom: basecro
```

**Comment by yihuang** [comment]
> maybe create a separate PR to fix workflow with submodule first
> and another PR related to non workflow changes after previous PR merged then run `/runsim`?

good idea, https://github.com/crypto-org-chain/chain-main/pull/387

**Review by tomtau — CHANGES_REQUESTED** [review]
I guess this shell hook doesn't need to clone the repos -- https://github.com/crypto-org-chain/chain-main/blob/master/pystarport/pystarport/convert.sh -- just use the submodule from the root dir? @leejw51crypto

**Comment by tomtau** [comment]
> I guess this shell hook doesn't need to clone the repos -- https://github.com/crypto-org-chain/chain-main/blob/master/pystarport/pystarport/convert.sh -- just use the submodule from the root dir? @leejw51crypto

and https://github.com/crypto-org-chain/chain-main/blob/master/nix/sources.json#L2 - @yihuang  ?

**Review by tomtau — APPROVED** [review]
sources.json are ok -- it's a different problem from what this PR is solving

**Comment by yihuang** [comment]
> sources.json are ok -- it's a different problem from what this PR is solving

I guess we can indeed remove the cosmos-sdk from sources.json, it's only used to build cosmovisor.

**Pull request merged**